packages/jest-haste-map/src/index.ts

```diff
@@ -1070,8 +1061,8 @@ export default class HasteMap extends EventEmitter {
 
     let dedupMap = hasteMap.map.get(moduleName);
 
-    if (dedupMap == null) {
-      dedupMap = Object.create(null) as {};
+    if (!dedupMap) {
+      dedupMap = Object.create(null) as ModuleMapItem;
```

Comment:
`dedupMap` does not have value yet, so `{}` is somewhat fine. But the value gets assigned just a couple of lines later. So it makes sense to type it with correct shape already here.